### ISSUE TITLE
Coffeescript highlighting is broken by a recent commit in this module

### ISSUE THREAD

**Issue by tremby** [ap/vim-css-color#63]
I'm highlighting coffeescript with https://github.com/kchmck/vim-coffee-script

I noticed today after updating all my vim modules that files were no longer highlighting the same way. In particular, with only vim-coffee-script enabled the following

```coffee
Classname
CONSTANT
```

renders in two different colours. On my screen (CLI, 256 colours, default colourscheme, bg=dark), `Classname` is green and `CONSTANT` is pink.

With vim-css-color also enabled, both highlight in pink.

I bisected vim-css-color, and the offending commit is this one: https://github.com/ap/vim-css-color/commit/469bf3f24b9808e9b8e817c8de2645407bcb7911

I'm not sure if this is a big in vim-css-color or vim-coffee-script. Any thoughts?

**Comment by ap** [comment]
Hmm, that should highlight CSS colours in Javascript strings and comments. `Classname` isn’t any kind of colour name so it shouldn’t be getting affected.

To debug this can you please place the cursor on each of the words and then run the following command, once with css-color enabled and once disabled? (Meaning 4 results.)

````vim
:echo map(synstack(line('.'), col('.')), 'synIDattr(v:val, "name")')
````

This outputs what syntax highlight groups were matched.

**Comment by tremby** [comment]
Without vim-css-color, each character in the first line (`Classname`) is `['coffeeObject']`. Each in the second line (`CONSTANT`) is `['coffeeConstant']`.

With vim-css-color, each character of both lines comes up as `['coffeeConstant']`. But a word consisting of only a single uppercase character comes up as `['coffeeObject']`. (This is the same without vim-css-color.)

**Comment by ap** [comment]
OK, I guess I have to install the plugin myself and track this down then.

**Comment by ap** [comment]
Well, adding the W3C colour keywords breaks the highlighting. I’ve not found out why yet. But I also noticed that the plugin currently doesn’t do much in Coffeescript buffers anyway, because even though they include the Javascript syntax, they mostly use their own syntax groups – and therefore end up with nothing much highlighted. So I took the cheap way out and just made it stay out of Coffeescript buffers.

I’ll look into adding explicit support for Coffeescript at some other point.
